Mathlib/Combinatorics/Optimization/ValuedCSP.lean

```diff
@@ -65,3 +67,156 @@ def ValuedCsp.Instance.evalSolution {Γ : ValuedCsp D C} {ι : Type*}
 def ValuedCsp.Instance.IsOptimumSolution {Γ : ValuedCsp D C} {ι : Type*}
     (I : Γ.Instance ι) (x : ι → D) : Prop :=
   ∀ y : ι → D, I.evalSolution x ≤ I.evalSolution y
+
+def Function.HasMaxCutPropertyAt (f : (Fin 2 → D) → C) (a b : D) : Prop :=
+  f ![a, b] = f ![b, a] ∧  -- `argmin f` is exactly `{ ![a, b] , ![b, a] }`
+    ∀ x y : D, f ![a, b] ≤ f ![x, y] ∧ (f ![a, b] = f ![x, y] → a = x ∧ b = y ∨ a = y ∧ b = x)
+
+def Function.HasMaxCutProperty (f : (Fin 2 → D) → C) : Prop :=
+  ∃ a b : D, a ≠ b ∧ f.HasMaxCutPropertyAt a b
+
+def FractionalOperation (D : Type _) (m k : ℕ) : Type* :=
```

Comment:
This looks like nonsense to me, you probably want
```suggestion
def FractionalOperation (D : Type*) (m k : ℕ) : Type _ :=
```
or if you want to be explicit
```suggestion
def FractionalOperation (D : Type u) (m k : ℕ) : Type u :=
```

Reply:
Thanks!